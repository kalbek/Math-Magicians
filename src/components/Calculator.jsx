import React, { useState } from 'react';
import keypads from './keypads.json';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [values, setValue] = useState({
    total: '0',
    next: null,
  });

  const onChange = (e) => {
    const calculated = calculate(values, e.target.innerText);
    setValue(calculated);
  };

  const { total, next } = values;
  const answer = next || total || '0';

  return (
    <>
      <section className="calculator flex">
        <div className="math-kid">
          <img
            src="https://th.bing.com/th/id/R.83950bc183d94ac68ff6e551a33f9dc2?rik=T1%2fhBDTPGZTTrg&riu=http%3a%2f%2fwww.scasd.org%2fcms%2flib5%2fPA01000006%2fCentricity%2fDomain%2f935%2fj0395714.gif&ehk=3PcQgIDiHi5f3%2fcf6q%2bYhR%2bH8FUQWkbMrzAo1dCkivE%3d&risl=&pid=ImgRaw&r=0"
            alt="math kid"
          />
        </div>
        <div className="container">
          <div className="display">{answer}</div>
          {keypads.map((item) => (
            <button
              key={item.id}
              name={item.name}
              type="button"
              className={item.className}
              onClick={onChange}
              value={item.text}
            >
              {item.text}
            </button>
          ))}
        </div>
      </section>
    </>
  );
};

export default Calculator;
