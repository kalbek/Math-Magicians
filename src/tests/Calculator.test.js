import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Calculator from '../components/Calculator';

test('It should render correctly', () => {
  const tree = render(
    <BrowserRouter>
      <Calculator />
    </BrowserRouter>,
  );
  expect(tree).toMatchSnapshot();
});
