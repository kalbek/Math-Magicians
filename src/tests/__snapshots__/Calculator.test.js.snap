// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="calculator flex"
      >
        <div
          class="flex-column math-kid"
        >
          <div
            class="do-math"
          >
            Let's do some math!
          </div>
          <div>
            <img
              alt="math kid"
              src="https://th.bing.com/th/id/R.83950bc183d94ac68ff6e551a33f9dc2?rik=T1%2fhBDTPGZTTrg&riu=http%3a%2f%2fwww.scasd.org%2fcms%2flib5%2fPA01000006%2fCentricity%2fDomain%2f935%2fj0395714.gif&ehk=3PcQgIDiHi5f3%2fcf6q%2bYhR%2bH8FUQWkbMrzAo1dCkivE%3d&risl=&pid=ImgRaw&r=0"
            />
          </div>
        </div>
        <div
          class="container"
        >
          <div
            class="display"
          >
            0
          </div>
          <button
            class="lighter"
            type="button"
            value="AC"
          >
            AC
          </button>
          <button
            class="lighter"
            type="button"
            value="+/-"
          >
            +/-
          </button>
          <button
            class="lighter"
            type="button"
            value="%"
          >
            %
          </button>
          <button
            class="orange"
            type="button"
            value="รท"
          >
            รท
          </button>
          <button
            class="lighter"
            type="button"
            value="7"
          >
            7
          </button>
          <button
            class="lighter"
            type="button"
            value="8"
          >
            8
          </button>
          <button
            class="lighter"
            type="button"
            value="9"
          >
            9
          </button>
          <button
            class="orange"
            type="button"
            value="x"
          >
            x
          </button>
          <button
            class="lighter"
            type="button"
            value="4"
          >
            4
          </button>
          <button
            class="lighter"
            type="button"
            value="5"
          >
            5
          </button>
          <button
            class="lighter"
            type="button"
            value="6"
          >
            6
          </button>
          <button
            class="orange"
            type="button"
            value="-"
          >
            -
          </button>
          <button
            class="lighter"
            type="button"
            value="1"
          >
            1
          </button>
          <button
            class="lighter"
            type="button"
            value="2"
          >
            2
          </button>
          <button
            class="lighter"
            type="button"
            value="3"
          >
            3
          </button>
          <button
            class="orange"
            type="button"
            value="+"
          >
            +
          </button>
          <button
            class="lighter zero-btn"
            type="button"
            value="0"
          >
            0
          </button>
          <button
            class="lighter"
            type="button"
            value="."
          >
            .
          </button>
          <button
            class="orange"
            type="button"
            value="="
          >
            =
          </button>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="calculator flex"
    >
      <div
        class="flex-column math-kid"
      >
        <div
          class="do-math"
        >
          Let's do some math!
        </div>
        <div>
          <img
            alt="math kid"
            src="https://th.bing.com/th/id/R.83950bc183d94ac68ff6e551a33f9dc2?rik=T1%2fhBDTPGZTTrg&riu=http%3a%2f%2fwww.scasd.org%2fcms%2flib5%2fPA01000006%2fCentricity%2fDomain%2f935%2fj0395714.gif&ehk=3PcQgIDiHi5f3%2fcf6q%2bYhR%2bH8FUQWkbMrzAo1dCkivE%3d&risl=&pid=ImgRaw&r=0"
          />
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="display"
        >
          0
        </div>
        <button
          class="lighter"
          type="button"
          value="AC"
        >
          AC
        </button>
        <button
          class="lighter"
          type="button"
          value="+/-"
        >
          +/-
        </button>
        <button
          class="lighter"
          type="button"
          value="%"
        >
          %
        </button>
        <button
          class="orange"
          type="button"
          value="รท"
        >
          รท
        </button>
        <button
          class="lighter"
          type="button"
          value="7"
        >
          7
        </button>
        <button
          class="lighter"
          type="button"
          value="8"
        >
          8
        </button>
        <button
          class="lighter"
          type="button"
          value="9"
        >
          9
        </button>
        <button
          class="orange"
          type="button"
          value="x"
        >
          x
        </button>
        <button
          class="lighter"
          type="button"
          value="4"
        >
          4
        </button>
        <button
          class="lighter"
          type="button"
          value="5"
        >
          5
        </button>
        <button
          class="lighter"
          type="button"
          value="6"
        >
          6
        </button>
        <button
          class="orange"
          type="button"
          value="-"
        >
          -
        </button>
        <button
          class="lighter"
          type="button"
          value="1"
        >
          1
        </button>
        <button
          class="lighter"
          type="button"
          value="2"
        >
          2
        </button>
        <button
          class="lighter"
          type="button"
          value="3"
        >
          3
        </button>
        <button
          class="orange"
          type="button"
          value="+"
        >
          +
        </button>
        <button
          class="lighter zero-btn"
          type="button"
          value="0"
        >
          0
        </button>
        <button
          class="lighter"
          type="button"
          value="."
        >
          .
        </button>
        <button
          class="orange"
          type="button"
          value="="
        >
          =
        </button>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
